src/Features/LanguageServer/Protocol/Api/InProcLanguageServer.cs

```diff
@@ -0,0 +1,404 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.ErrorReporting;
+using Microsoft.CodeAnalysis.LanguageServer.Handler;
+using Microsoft.CodeAnalysis.Shared.TestHooks;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Roslyn.Utilities;
+using StreamJsonRpc;
+
+using LSP = Microsoft.VisualStudio.LanguageServer.Protocol;
+
+namespace Microsoft.CodeAnalysis.LanguageServer.Api
+{
+    internal class InProcLanguageServer : IAsyncDisposable
```

Comment:
The "InProc" seems a bit odd naming here, given well, it's an object, so of course it's in-proc.